server/src/main/java/io/deephaven/server/session/SessionServiceGrpcImpl.java

```diff
@@ -290,24 +314,144 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(final ServerCall<Re
                 }
             }
 
+            // Lookup the session using Flight Auth 2.0 token.
             final String token = metadata.get(SESSION_HEADER_KEY);
             if (session == null && token != null) {
                 try {
                     session = service.getSessionForAuthToken(token);
                 } catch (AuthenticationException e) {
-                    try {
-                        call.close(Status.UNAUTHENTICATED, new Metadata());
-                    } catch (IllegalStateException ignored) {
-                        // could be thrown if the call was already closed. As an interceptor, we can't throw,
-                        // so ignoring this and just returning the no-op listener.
+                    return onFailure.get();
+                }
+            }
+
+            // find the auth method
+            final String serviceName = call.getMethodDescriptor().getServiceName();
```

Comment:
I'm beginning to think we should have a singular generic entrypoint instead of the per-service way it's structured right now. `authorization.check(authContext, call.getMethodDescriptor(), message)`. If we want per-service classes, that could be an implementation detail IMO.

By allowing the hooking of a single entrypoint, instead of N, it's a much easier job to configure IMO.

Reply:
We can probably still do this, but it looks like I'm going to skip this iteration in favor for the type safe interfaces for now since I was able to eliminate the reflection or even an additional layer of indirection.